src/main/java/com/direwolf20/buildinggadgets/common/tools/ExchangingModes.java

```diff
@@ -16,26 +15,154 @@
+            EnumFacing playerFacing = player.getHorizontalFacing();
+            base = hit.offset(playerFacing.getOpposite(), radius);
+            extension = playerFacing;
+        } else {
+            base = hit.down(radius);
+            extension = EnumFacing.UP;
+        }
+
+        for (int i = 0; i < range; i++) {
+            BlockPos pos = base.offset(extension, i);
+            if (isReplaceable(world, pos, current, target, fuzzyMode)) {
+                coordinates.add(pos);
+            }
+        }
+
+        return coordinates;
+    }),
+    HorizontalColumn((world, player, hit, sideHit, tool) -> {
+        List<BlockPos> coordinates = new ArrayList<>();
+        IBlockState target = GadgetUtils.getToolActualBlock(tool);
+        IBlockState current = world.getBlockState(hit);
+        int range = GadgetUtils.getToolRange(tool);
+        int radius = (range - 1) / 2;
+        boolean fuzzyMode = GadgetGeneric.getFuzzy(tool);
+
+        EnumFacing playerFacing = player.getHorizontalFacing();
+        EnumFacing extension = playerFacing.rotateY();
+        BlockPos base = hit.offset(extension.getOpposite(), radius);
+
+        for (int i = 0; i < range; i++) {
+            BlockPos pos = base.offset(extension, i);
+            if (isReplaceable(world, pos, current, target, fuzzyMode)) {
+                coordinates.add(pos);
+            }
+        }
+
+        return coordinates;
+    }),
+    Grid((world, player, hit, sideHit, tool) -> {
+        List<BlockPos> coordinates = new ArrayList<>();
+        IBlockState target = GadgetUtils.getToolActualBlock(tool);
+        IBlockState current = world.getBlockState(hit);
+        int range = GadgetUtils.getToolRange(tool) + 1;
+        boolean fuzzyMode = GadgetGeneric.getFuzzy(tool);
+
+        for (int x = range * -7 / 5; x <= range * 7 / 5; x++) {
+            for (int z = range * -7 / 5; z <= range * 7 / 5; z++) {
+                if (x % (((range - 2) % 6) + 2) == 0 && z % (((range - 2) % 6) + 2) == 0) {
+                    BlockPos pos = new BlockPos(hit.getX() + x, hit.getY(), hit.getZ() + z);
+                    if (isReplaceable(world, pos, current, target, fuzzyMode)) {
+                        coordinates.add(pos);
+                    }
+                }
+            }
+        }
+
+        return coordinates;
+    });
+
+    private final String displayName;
+    private final IBuildingMode modeImpl;
+
+    ExchangingModes(IBuildingMode modeImpl) {
+        this.displayName = GadgetGeneric.formatName(name());
+        this.modeImpl = modeImpl;
+    }
+
+    @Override
+    public String toString() {
+        return displayName;
+    }
+
+    public ExchangingModes next() {//TODO unused
+        ExchangingModes[] values = values();
+        return values[(this.ordinal() + 1) % values.length];
+    }
+
+    public static IBuildingMode byName(String name) {
+        return Arrays.stream(values())
+                .filter(mode -> mode.name().equals(name))
```

Comment:
I know this is done evrywhere in the mod, but one should never rely upon a enum name (see effective Java for reference), so could we use the id's decribed internally as keys?

Also IBuildingMode should gain a translate Method, which would then be called for Display...

Reply:
It is `Enum#name()` from Java, which is the same name in code.

Reply:
I know that. But using Enum#name imposes implementation restrictions as well as producing an very unflexible API. BG doing that in the past was bad style, which is why I wanted to know whether changing that would fall into the scope of this PR.